From 8483afe9dd072489b16a1d8c1e78d0f179219ca3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Arve=20Hj=C3=B8nnev=C3=A5g?= <arve@android.com>
Date: Mon, 4 Mar 2013 19:14:10 -0800
Subject: [PATCH 0374/5965] usb: gadget: android: move init to late_initcall
 for now
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gserial_alloc_line crashes when called from module_init

Signed-off-by: Arve Hjønnevåg <arve@android.com>
---
 drivers/usb/gadget/android.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index 60072f3af96c..f5b00c122585 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -1472,7 +1472,7 @@ err_dev:
 	class_destroy(android_class);
 	return err;
 }
-module_init(init);
+late_initcall(init);
 
 static void __exit cleanup(void)
 {
-- 
2.19.0

